### PR TITLE
Update: Use PaceJS instead of NProgress

### PR DESCRIPTION
**Pull request by Halyul**
**What kind of change does this PR introduce?** (check one with "x")

- [ ] Bug fix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?** (check one with "x")

- [x] Yes
- [ ] No

____

**Description**

Use PaceJS instead of NProgress.
And this commit includes 14 styles (in folder source/css/pace), but two of which (center-atom, center-radar) have some problem, needed to be fixed. (I have no idea about that)

____

**Verification steps**

No verification steps.

### CODE REVIEW

File: layout/_partial/head.ejs [hunk 1]
```diff
@@ -83,6 +83,9 @@
     <%- css("css/style.min") %>
     <%- partial("_partial/config_css") %>
     <%- js("js/jquery.min") %>
+	<!--Import pace -->
+	<%- partial("_partial/import_pace") %>
```
Comment (pidupuis) [inline layout/_partial/head.ejs]: - Please remove comment
- Please fix tab

File: _config.yml [hunk 2]
```diff
@@ -32,8 +32,7 @@ uiux:
     hyperlink_color: "#00838F"
     button_color: "#757575"
     android_chrome_color: "#0097A7"
-    nprogress_color: "#FF4081"
-    nprogress_buffer: "800"
+    buffer: "800"
```
Comment (viosey) [inline _config.yml]: Color preference needed

File: _config.yml [hunk 3]
```diff
@@ -78,6 +77,14 @@ img:
...
 fonts: Roboto, "Helvetica Neue", Helvetica, "PingFang SC", "Hiragino Sans GB", "Microsoft YaHei", "微软雅黑", Arial, sans-serif
 
+# Pace Settings
+# Pace is automatic page load progress bar
+#  You can find demos on http://github.hubspot.com/pace/docs/welcome
+#     style available values:
+#       barber-shop, big-counter, bounce, center-circle,
+# 		center-simple, corner-indicator, fill-left, flash, flat-top, loading-bar, macosx
+pace: 
+
```
Comment (pidupuis) [inline _config.yml]: - You should put a default value similar to nprogress style

File: layout/_partial/import_pace.ejs [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<%- js("js/pace.min") %>
```
Comment (pidupuis) [inline layout/_partial/import_pace.ejs]: - Please use single quotes (in the entire file)

File: layout/_partial/import_pace.ejs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+<% if(theme.pace == 'barber-shop') { %>
+   		 <%- css("css/pace/barber-shop") %>
+	<% } else if(theme.pace == 'big-counter') { %>
```
Comment (pidupuis) [inline layout/_partial/import_pace.ejs]: - Please use strict equality `===` (in the entire file)

File: layout/_partial/head.ejs [hunk 6]
```diff
@@ -83,6 +83,9 @@
     <%- css("css/style.min") %>
     <%- partial("_partial/config_css") %>
     <%- js("js/jquery.min") %>
```
Comment (pidupuis) [inline layout/_partial/head.ejs]: A js import shouldn't be inside css block import...

**Comment by Halyul** [comment]
Yep..I finished this in a hurry...It may be not so prefect

File: _config.yml (hunk 3 above)
Comment (Halyul) [inline _config.yml]: if users dont configure it, it will use minimal style by default

Comment (pidupuis) [inline _config.yml]: That's ok then :+1:

**Comment by pidupuis** [comment]
@Halyul No worry, that's what reviewers are for ^^

File: layout/_partial/head.ejs (hunk 1 above)
Comment (Halyul) [inline layout/_partial/head.ejs]: Well, which comment should I remove..? Line 86?

**Comment by Halyul** [comment]
Do not merge! I will use canary branch to create a new pr. THIS IS BASED ON MASTER

**Comment by viosey** [comment]
migrate to https://github.com/viosey/hexo-theme-material/tree/feat/progressbar

**Pull request closed without merging**